OpenRA.Mods.Common/Traits/Turreted.cs

```diff
@@ -85,32 +86,30 @@ public static Func<int> TurretFacingFromInit(IActorInitializer init, int def, st
 		}
 
 		public Turreted(ActorInitializer init, TurretedInfo info)
+			: base(info)
 		{
 			this.info = info;
 			TurretFacing = TurretFacingFromInit(init, info.InitialFacing, info.Turret)();
 		}
 
 		public void Created(Actor self)
 		{
-			attack = self.TraitOrDefault<AttackTurreted>();
+			attack = self.TraitsImplementing<AttackTurreted>().ToArray();
```

Comment:
Okay, now I see what you mean.

AttackTurreted should filter out to the turrets which are appropriate for it's respective armaments. I'd say right now that means AttackTurreted.Turrets would be set up in RulesetLoaded, because Requires would introduce a cyclic dependency and longterm to shift the turret handling from armaments to the attack traits as well (which will probably be a mess on it's own).